torch/csrc/jit/passes/quantization.cpp

```diff
@@ -459,16 +459,16 @@ class QuantizeHelper {
   }
 
  private:
-  const script::Module& module_;
-  std::vector<std::string> observer_modules_to_remove_;
+  script::Module& module_;
   std::vector<Node*> nodes_to_destroy_;
 };
 
 void QuantizeHelper::removeObserver(
     Value* v,
     const std::string& observer_name) {
   // remove observer_module
-  observer_modules_to_remove_.push_back(observer_name);
+  module_.module_object()->unsafeRemoveAttr(observer_name);
```

Comment:
Every `unsafeRemove` has a O(N) complexity where N is the number of slots after the one we're removing. In our case it will be a number of observers. We will do that for each observer, so the total complexity of the algorithm would be O(N^2). Would it be possible to change it to be linear (e.g. delete all of them starting from the last in the very end)?

Reply:
Sure

Reply:
it will complicate the code a bit tho